artemis-core-client/src/main/java/org/apache/activemq/artemis/api/core/management/CoreNotificationType.java

```diff
@@ -47,7 +47,8 @@
    SESSION_CREATED(26),
    SESSION_CLOSED(27),
    MESSAGE_DELIVERED(28),
-   MESSAGE_EXPIRED(29);
+   MESSAGE_EXPIRED(29),
+   CONNECTION_CONNECTED(30);
```

Comment:
because this is in core, this must be implemented in a generic way for ALL protocols.

Reply:
Going through and running this, and running with notification plugin, i notice every time this occurs, a SESSION_CREATED is also sent. 

digging deeper and looking going through the logic of the connect in MQTTConnectionManager it seems this actually connected in the MQTTConnectionManager maps over to CORE as a server session (e.g. connect == create a serversession), which means we already get a SESSION_CREATED notification with that plugin, so is it actually we want to ensure SESSION_CREATED notificaions are not reliant on plugin, e.g. bring that in... 

populating.
session.getRemotingConnection().getClientID()

Reply:
So this ties in with the fact you want to ensure a single connected session (its not actually at connection level, its actually at the session level).

Reply:
I am here to make a unified reply to your comments.

In fact, this is only the mqtt protocol need, other protocols do not.Either way, it doesn't look too good in CORE, it just tries to make it look reasonable and more readable.
```
   /**
    * Called when all connection related operations are completed.
    */
   default void connected() throws Exception {
```
I use `connected` as default not a interface method，Because we need to send cluster-notifation when the mqtt protocol connection is initialized, But other protocols don't need this method to do anything, use a empty default methods I think is appropriate.And all protocol connections have this connection completion status, and other protocols have the possibility to use this method. As for the logic that is only needed for MQTT to add a listener or callback, I think it is over-designed.

I also find SESSION_CREATED  and CONNECTION_CREATED, but i can not use them, because we need clientId, and clientId read from client message. The clientId is read from the client's connection message. The logic for reading the clientId cannot be placed in the CORE code， when create a connection or create a session.

For using callbacks or listeners, I still prefer to use the default method connected. Unless you have a better design to let me change my mind. 
the clientId is read from the client message, it seems that the message can only be sent after the protocol connection has been processed. ( MQTT protocol neet to process the CONNECT request packet, some protocols do not have a CONNECT request packet)

Reply:
re

"The clientId is read from the client's connection message. The logic for reading the clientId cannot be placed in the CORE code， when create a connection or create a session."

the clientid is set on the connection BEFORE createServerSession as such it will be available when the SESSION_CREATED notification is created, 

MQTTConnectionManager lines 78, 79
```
 session.getConnection().setClientID(clientId);
 ServerSessionImpl serverSession = createServerSession(username, password);
```
just need to populate the extra field which would be available at the time via session.getRemotingConnection().getClientID() .

Reply:
```
if (hasBrokerSessionPlugins()) {
    callBrokerSessionPlugins(plugin -> plugin.afterCreateSession(session));
}
```
SESSION_CREATED will only be sent if BrokerSessionPlugins is turned on in the configuration file.
Also the notifications of CONNECTION_CREATE and SESSIONTION_CREATE will be filtered and will not be sent to other nodes in the cluster.You can see `ClusterConnectionBridge.setupNotificationConsumer()` method, where the cluster-connection filter is set .
This is why I want to modify the filtering rules, CONNECTION_CREATE, SESSIONTION_CREATE and other connection-related notifications, only can use the `ManagementNotificationAddress` address to route messages.

Reply:
```
private void handleNotificationMessage(ClientMessage message, CoreNotificationType ntype) throws Exception {
         // TODO - optimised this by just passing int in header - but filter needs to be extended to support IN with

         switch (ntype) {
            case BINDING_ADDED: {
               doBindingAdded(message);

               break;
            }
            case BINDING_REMOVED: {
               doBindingRemoved(message);

               break;
            }
            case CONSUMER_CREATED: {
               doConsumerCreated(message);

               break;
            }
            case CONSUMER_CLOSED: {
               doConsumerClosed(message);

               break;
            }
            case PROPOSAL: {
               doProposalReceived(message);

               break;
            }
            case PROPOSAL_RESPONSE: {
               doProposalResponseReceived(message);
               break;
            }
            case UNPROPOSAL: {
               doUnProposalReceived(message);
               break;
            }
            default: {
               throw ActiveMQMessageBundle.BUNDLE.invalidType(ntype);
            }
         }
      }
```

Only these notifications will pass between cluster nodes